src/App.jsx

```diff
@@ -14,45 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'alphabeticly';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, isReversed) {
+  const preperedGoods = [...goods];
+
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        default:
+          return 0;
+
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const goods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
+  const reset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  const reverseGoods = () => setIsReversed(!isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
+          onClick={() => setSortField(SORT_FIELD_NAME)}
```

Comment:
good practice to make one function factory for several buttons with same logic.
```
const handleSort = (sort) => () => setSortField(sort);

```
```suggestion
          onClick={handleSort(SORT_FIELD_NAME)}
```